Contenido bajo licencia Creative Commons BY 4.0 y código bajo licencia MIT. © Nicolás Guarín-Zapata 2026.

# Integración numérica

In [9]:
import numpy as np
from scipy.integrate import quad, fixed_quad, quadrature
import matplotlib.pyplot as plt

La rutina más general para la integración númerica es ``scipy.integrate.quad()``

In [10]:
res, err = quad(np.sin, 0, np.pi/2)


``res`` contiene la respuesta de la integral

In [11]:
print(res)

0.9999999999999999


``err`` contiene la estimación del error en la integración numérica

In [12]:
print(err)

1.1102230246251564e-14


Además, tenemos las rutinas ``fixed_quad`` y ``quadrature``

Usando ``fixed_quad`` realizamos una integral por el método de Gauss
para una cantidad de puntos de Gauss deseada

In [13]:
fixed_quad(lambda x: x**7 - 4, 0, 1, n=4)

(-3.875, None)

Y vemos que este es el valor exacto de la integral

$$ \int_{0}^1 x^7 - 4 \mathrm{d}x = \left[\frac{x^8}{8} - 4 x\right]_{0}^{1} = -3.875$$

In [14]:
1/8 - 4

-3.875

La rutina ``quadrature`` usa un método de Gauss adaptativo, por lo tanto
nos garantiza que la precisión de la integral está dada por un valor prefijado.

Si repetimos la integral anterior, obtenemos

In [15]:
quadrature(lambda x: x**7 - 4, 0, 1)

(-3.8750000000000004, 4.440892098500626e-16)

Lo cual nos dice que la integral es aproximadamente -3.8750000000000004 y que el
error es del orden de 4.4408920985006262e-16